mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/assets/android/AndroidAssetManager.kt

```diff
@@ -0,0 +1,17 @@
+package ch.epfl.sdp.mobile.infrastructure.assets.android
+
+import android.content.Context
+import ch.epfl.sdp.mobile.infrastructure.assets.AssetManager
+
+/**
+ * Represents the Android Assets Manager, which allows to load assets in different manners
+ *
+ * @param context The Android [Context] used to load assets
+ */
+class AndroidAssetManager(
+    private val context: Context,
+) : AssetManager {
+  override fun readText(path: String): String? {
+    return context.assets.open(path).reader().readText()
+  }
```

Comment:
You may want to catch some exceptions, and close the `Reader` once all the content has been read.

1. `runCatching` ensures you catch the possible exceptions.
2. `use` ensures you properly close the `Reader` when it's not needed anymore.

```kotlin
runCatching { reader.use { it.readText() } }.getOrNull()
```

Reply:
https://github.com/epfl-SDP/android/pull/300/commits/641143be45dbc561470b284be9873bc46e4cfc33